csi/controller.go

```diff
@@ -980,6 +1008,99 @@ func (s *OsdCsiServer) CreateSnapshot(
 		},
 	}, nil
 }
+func (s *OsdCsiServer) getCloudBackupClient() error {
+	if s.cloudBackupClient == nil {
+		// Get grpc connection
+		conn, err := s.getConn()
+		if err != nil {
+			return status.Errorf(
+				codes.Unavailable,
+				"Unable to connect to SDK server: %v", err)
+		}
+		s.cloudBackupClient = api.NewOpenStorageCloudBackupClient(conn)
+	}
+	return nil
+}
+func (s *OsdCsiServer) createCloudBackup(
+	ctx context.Context,
+	req *csi.CreateSnapshotRequest,
+) (*csi.CreateSnapshotResponse, error) {
+	err := s.getCloudBackupClient()
+	if nil != err {
+		return nil, err
+	}
+
+	// Get any labels passed in by the CO
+	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetParameters())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
+	}
+
+	backupID := req.GetName()
+	// Create snapshot
+	_, err = s.cloudBackupClient.Create(ctx, &api.SdkCloudBackupCreateRequest{
+		VolumeId: req.GetSourceVolumeId(),
+		TaskId:   backupID,
+		Labels:   locator.GetVolumeLabels(),
+	})
+
+	if err != nil {
+		errStatus, ok := status.FromError(err)
+		if ok && errStatus.Code() == codes.NotFound {
+			// Return a non-final Aborted error, as the PVC may not have been created yet. Otherwise
+			// the CSI Snapshotter will not issue retries:
+			// https://github.com/kubernetes-csi/external-snapshotter/blob/v6.0.1/pkg/sidecar-controller/snapshot_controller.go#L656-L678
+			return nil, status.Errorf(codes.Aborted, "Volume id %s not found: %v", req.GetSourceVolumeId(), err)
+		}
```

Comment:
This was a special case for local snapshots. We can remove this from cloud snapshots.